src/poller.rs

```diff
@@ -1,38 +1,91 @@
-use std::path::Path;
+use std::fs;
+use std::path::{Path, PathBuf};
 use std::process;
 use std::time::Instant;
 
 use tokio::time;
 use tracing::{error, info};
 
-use crate::config::Config;
+use crate::config::{Config, WatchConfig};
 use crate::log::Operation;
 use crate::snapshots;
 
+/// Checks the provided `child_path` is a directory.
+/// If either `includes` or `excludes` are set,
+/// checks whether the path is included/excluded respectively.
+fn is_valid_directory(base_path: &Path, child_path: &PathBuf, value: &WatchConfig) -> bool {
+    if !child_path.is_dir() {
+        return false;
+    }
+
+    let includes = &value.include;
+    let excludes = &value.exclude;
+
+    if includes.len() > 0 {
+        let relative = child_path.strip_prefix(base_path).unwrap().to_str().unwrap();
```

Comment:
See if you can get rid of the `.unwrap()` calls. I think returning `false` would be an appropriate recovery in both cases.